### DISCUSSION TITLE
Is there any way to run fish from a folder ?

### DISCUSSION THREAD

**Discussion by Xoffio** [fish-shell/fish-shell#9435] [Q&A] [answered]
Hi!

I downloaded the rpm file from the fish website, exported the compiled files with
```bash
rpm2cpio fish-3.5.1-1.2.x86._64.rpm | cpio -idv
```

and now i can execute fish by typing 

```
./usr/bin/fish
```

Fish opens without any error but the theme does not show and the autocomplete does not work as expected.. I am installing it this way because i dont have root access to the system. I think this can be solve by adding some paths? 

Have anyone tried something like this before ? Do you know if there is any chance to run it this way? thanks !

**Comment by faho** [comment, marked as answer]
The rpm has been built to use the system paths. It has been configured to be installed into /usr, so fish ends up in /usr/bin/fish and the associated files end up in /usr/share/fish/.

That's how rpms typically work. You're not really gonna be happy just unpacking them like that.

You can build fish yourself with other files, or you could try [this appimage a user has produced](https://github.com/mliszcz/fish-shell/releases/tag/fish-3.5.1-x86_64). (building an official appimage is also our longer term goal for having an easy-to-install single-file fish, but this one is very much unofficial)

**Reply by Xoffio** [reply]
I see... I tried to compile it my self but the computer have cmake 2.4 and I can't upgrade it ill try that appimage if there is not other other option.. 

Just for curiosity... If that user already made an appimage what is the challange you are having creating one? does it mean that the appimage created by that user is not stable? or insecure? 

Thanks for taking the time and for the contribution of such of useful tool!

**Reply by faho** [reply]
>but the computer have cmake 2.4 and I can't upgrade it

Cmake has [binary packages](https://cmake.org/download/) for x86_64 and aarch64. These are ones you can just unpack and run.

>what is the challange you are having creating one?

The issue with the appimage is mainly that it needs testing (where does or does it not work? Old distros, new distros, distros using musl?), and we need to figure out where and how it'll be built - that version statically links ncurses and glibc, which is typically a big no-no (and we don't usually support static linking because it only leads to problems), and it's built by a weird pipeline running a CentOs container on Github Actions' Ubuntu machines.

**Reply by Xoffio** [reply]
@faho Thank you for the Cmake tip. I tried that but when doing `make` it will thought some error that I believe are related to C++11 Do you know if i can try one more thing or i should just go for the appimage ?

**Reply by faho** [reply]
Ah, your compiler is *that* outdated - *almost 10 years* (gcc 4.8.1 had full C++11 support, released march 2013).

Try the appimage but truth be told your system might be too old for even that.

I think without getting someone to upgrade the system, get you a newer compiler (RHEL has [developer toolsets](https://developers.redhat.com/products/developertoolset/overview) that can install newer toolchains in parallel) or install the fish package for you you might be in a world of pain.
